pkg/resolution/resolver/git/resolver.go

```diff
@@ -422,7 +422,15 @@ func populateDefaultParams(ctx context.Context, params map[string]string) (map[s
 		return nil, fmt.Errorf("missing required git resolver params: %s", strings.Join(missingParams, ", "))
 	}
 
-	// TODO(sbwsg): validate repo url is well-formed, git:// or https://
+	if params[urlParam] != "" {
+		if !strings.HasPrefix(params[urlParam], "git://") &&
```

Comment:
NITs: 
- "file://" should be valid too for testing purposes. 
- ssh supports the compact format <user>@... as well. If we don't want to support that, we need to document it.
- I often use "git+ssh://" which is a valid format too

See: https://git-scm.com/book/en/v2/Git-on-the-Server-The-Protocols#:~:text=Git%20can%20use%20four%20distinct,Shell%20(SSH)%20and%20Git.

Reply:
I'm not sure on `foo@bar:` - it's not as trivial to validate. But I'll definitely add `file://` and `git+ssh://`.